api/models.py

```diff
@@ -163,11 +166,15 @@ def conformity(self):
             return None
         highstate_ret = last_highstate.loaded_ret()
         for state in highstate_ret["return"]:
-            # Flat out error(return is a string)
-            if isinstance(highstate_ret["return"], list):
-                return False
-            # One of the state is not ok
-            if not highstate_ret["return"][state]["result"]:
+            try:
+                # Flat out error(return is a string)
+                if isinstance(highstate_ret["return"], list):
+                    return False
+                # One of the state is not ok
+                if not highstate_ret["return"][state]["result"]:
+                    return False
+            except Exception as e:
```

Comment:
yes i would assume that too.

tbh id rather do .get extractions and set sane defaults?

will update when i get a sec

Reply:
also its good practice to log at exceptions. otherwise you never catch things that you never know about.

Reply:
_also its good practice to log at exceptions. otherwise you never catch things that you never know about._

Thanks for the reminder.

_.. why is there no logging? first time ive seen a design decision around that?_

Lack/Cost of time mostly. It also doesn't help that there's 3 different ways to install Alcali currently(from pip, using docker or from sources). We can discuss adding logs in a different issue (as you can see  [here](https://github.com/latenighttales/alcali/blob/69f9cd197ac8603d2944226ba89274e569d0840d/config/settings.py#L151) it was already on my mind).

For now, let's just fix the related issue.